src/scenes/ObservationsRouter/adhoc/index.tsx

```diff
@@ -1,113 +1,170 @@
 
 export default function ObservationMonitoringPlot(): JSX.Element | undefined {
-  const { plantingSiteId, observationId, plantingZoneName, monitoringPlotId } = useParams<{
+  const params = useParams<{
     plantingSiteId: string;
     observationId: string;
     plantingZoneName: string;
     monitoringPlotId: string;
   }>();
+
+  const plantingSiteId = Number(params.plantingSiteId);
+  const observationId = Number(params.observationId);
+  const plantingZoneName = params.plantingZoneName;
+  const monitoringPlotId = Number(params.monitoringPlotId);
+
   const defaultTimeZone = useDefaultTimeZone();
   const navigate = useSyncNavigate();
   const theme = useTheme();
   const { isMobile } = useDeviceInfo();
   const { activeLocale } = useLocalization();
-  const allObservationsResults = useAppSelector(selectObservationsResults);
-  const observationsResults = useMemo(() => {
-    if (!allObservationsResults || !plantingSiteId) {
-      return [];
+
+  const { species } = useSpeciesData();
+  const { plantingSite, adHocObservationResults, observationResults } = usePlantingSiteData();
+  const [plantingZoneResult, setPlantingZoneResult] = useState<ObservationPlantingZoneResultsPayload>();
+  const [plantingSubzoneResult, setPlantingSubzoneResult] = useState<ObservationPlantingSubzoneResultsPayload>();
+  const [monitoringPlotResult, setMonitoringPlotResult] = useState<ObservationMonitoringPlotResultsPayload>();
+
+  const result = useMemo(() => {
+    if (!Number.isNaN(Number(observationId))) {
+      return (
+        observationResults?.find((_result) => _result.observationId === Number(observationId)) ??
+        adHocObservationResults?.find((_result) => _result.observationId === Number(observationId))
+      );
     }
-    return allObservationsResults?.filter((observationResult) => {
-      const matchesSite =
-        plantingSiteId !== '-1' ? observationResult.plantingSiteId.toString() === plantingSiteId : true;
-      return matchesSite;
-    });
-  }, [allObservationsResults, plantingSiteId]);
-
-  const monitoringPlot = useAppSelector((state) =>
-    selectObservationMonitoringPlot(
-      state,
-      {
-        plantingSiteId: Number(plantingSiteId),
-        observationId: Number(observationId),
-        plantingZoneName,
-        monitoringPlotId: Number(monitoringPlotId),
-      },
-      defaultTimeZone.get().id
-    )
-  );
+  }, [observationResults, adHocObservationResults, observationId]);
 
-  const plantingSite = useAppSelector((state) => selectPlantingSite(state, Number(plantingSiteId)));
+  useEffect(() => {
+    if (result) {
+      result.plantingZones.forEach((zone) =>
+        zone.plantingSubzones.forEach((subzone) =>
+          subzone.monitoringPlots.forEach((plot) => {
+            if (plot.monitoringPlotId === monitoringPlotId) {
+              setPlantingZoneResult(zone);
+              setPlantingSubzoneResult(subzone);
+              setMonitoringPlotResult(plot);
+            }
+          })
```

Comment:
I think a `return` should be added after the state setters so that the rest of the loops are not evaluated since they won't ever be true for the monitoringPlotId (I believe)